src/common/libjob/list.c

```diff
@@ -59,23 +67,30 @@ flux_future_t *flux_job_list_inactive (flux_t *h,
 {
     flux_future_t *f;
     json_t *o = NULL;
+    json_t *c = NULL;
     int saved_errno;
 
     if (!h || max_entries < 0 || since < 0. || !attrs_json_str
            || !(o = json_loads (attrs_json_str, 0, NULL))) {
         errno = EINVAL;
         return NULL;
     }
+    if (!(c = json_pack ("{ s:[ {s:[f]}, {s:[i]} ] }",
+                         "and",
+                         "since", since,
+                         "states", FLUX_JOB_STATE_INACTIVE))) {
+        json_decref (o);
+        errno = ENOMEM;
+        return NULL;
+    }
     if (!(f = flux_rpc_pack (h, "job-list.list", FLUX_NODEID_ANY, 0,
-                             "{s:i s:f s:i s:i s:i s:o}",
+                             "{s:i s:o s:o}",
                              "max_entries", max_entries,
-                             "since", since,
-                             "userid", FLUX_USERID_UNKNOWN,
-                             "states", FLUX_JOB_STATE_INACTIVE,
-                             "results", 0,
-                             "attrs", o))) {
+                             "attrs", o,
+                             "constraints", c))) {
```

Comment:
It looks like there's a mix of `constraint` vs `constraints` throughout this file. What is the difference? It wasn't obvious reading the diff anyway...

Reply:
That seems to just be an inconsistent I picked somewhat at random.  I'll correct.

Reply:
whew, good catch, at first I was like "how do the tests even work?" only to realize "oh yeah, we deprecated these functions", so I guess its not really tested at all.